chains/evm/contracts/ccvs/RampProxy.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+import {Ownable2StepMsgSender} from "@chainlink/contracts/src/v0.8/shared/access/Ownable2StepMsgSender.sol";
+
+/// @notice RampProxy enables upgrades to CCVRamps without breaking existing references in token pools, receivers, and apps.
+/// The address of this contract will be referenced in the following places:
+///   - Users / apps will specify required and optional CCVs as part of ccipSend extraArgs.
+///   - Token pools will specify required and optional CCVs on both source and destination.
+///   - Receiver contracts will specify required and optional CCVs on destination.
+///   - CCVProxy will specify default and mandated CCVs for each destination.
+///   - CCVAggregator will specify default and mandated CCVs for each source.
+/// Each of these references should be to a RampProxy contract, not a CCVRamp directly.
+/// @dev On source, the CCVProxy will forward requests (i.e. getFee, forwardToVerifier) through this contract to the required CCVRamp.
+/// The same applies on destination. The CCVAggregator will forward requests (i.e. verifyMessage) through this contract to the required CCVRamp.
+/// To support this proxy, all future CCVRamp interfaces must have originalCaller defined as the first arg to each method.
+contract RampProxy is Ownable2StepMsgSender {
```

Comment:
CCVProxy and CCVProxy should be renamed to OnRamp, but we can make that switch in a future rename PR, this one is already too large.